src/components/AutoCompleteInput.js

```diff
@@ -67,28 +63,28 @@ class AutoCompleteInput extends React.PureComponent {
       this.state.currentTrigger
     ];
     this.props.openSuggestions(title, component);
-    // console.log('start TRACKING');
   };
 
   stopTracking = () => {
     this.isTrackingStarted = false;
-    // console.log('STOP TRACKING');
     this.props.closeSuggestions();
   };
 
-  updateSuggestions(q) {
+  updateSuggestions = async (q) => {
     const triggers = this.props.triggerSettings;
-    this.props.updateSuggestions({
-      data: triggers[this.state.currentTrigger].dataProvider(
-        q,
-        this.state.text,
-      ),
-      onSelect: this.onSelectSuggestion,
-    });
-  }
+    await triggers[this.state.currentTrigger].dataProvider(
+      q,
+      this.state.text,
+      (data) => {
+        this.props.updateSuggestions({
```

Comment:
I was wondering if we shouldn't check if the query you are getting the response for is still relevant before updating it here. If the responses when querying members take a while to come, even with the debounce it could be that the user already changed the mentioned text, in which case they'll see mentions that are no longer relevant.

Reply:
Good point!! Let me see what I can do about it

Reply:
@vicnicius have added the check!

Reply:
👌